app/controllers/api/v1/post_comments_controller.rb

```diff
@@ -0,0 +1,36 @@
+class Api::V1::PostCommentsController < ApplicationController
+    def create
+        @comment = PostComment.new(comment_params)
+        @comment.user = current_user
+
+        if @comment.save
+            render json: {status: 'Successful', message: 'Comment added'}, status: 201
+        else
+            render json: {status: 'Failure', message: @comment.errors.full_messages}, status: 422
+        end
+    end
+
+    def destroy
+        @post = Post.find(params[:id])
+        @comment = @post.post_comments.find_by(id: params[:comment_id], user_id: current_user.id)
+
+        if @comment
+            if @comment.destroy
+                render json: { status: 'Success', message: 'Comment deleted' }, status: 200
+            else
+                render json: { status: 'Failure', message: 'Failed to delete comment' }, status: 422
+            end
+        else
+            render json: { status: 'Failure', message: 'Comment not found' }, status: 404
+        end
+    end
+
+    def index
+        @comment = PostComment.where(post_id: params[:id])
+        render json: { status: 'Success', message: 'Successful', data: @comment }, status: 200
+    end
```

Comment:
- There are no params passed to the index route. I understand why you used the show action in the first place. You can change back to the show action.

Reply:
Shebi you see! No dey follow me argue next time! wicked reviewer